### PR TITLE
feat(nbstore): init

### PR DESCRIPTION
**Pull request by forehalo**
TODO

- [x] basic
- [x] storages
- [x] producer/consumer
- [x] operation pattern
- [x] events
- [x] worker
- [x] readme
- [x] peer dependencies

### CODE REVIEW

**Comment by forehalo** [comment]
* **#7639** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/7639?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 2  dependent PRs ([#8539](https://github.com/toeverything/AFFiNE/pull/8539) <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/8539?utm_source=stack-comment-icon" target="_blank">Graphite</a>, [#8808](https://github.com/toeverything/AFFiNE/pull/8808) <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/8808?utm_source=stack-comment-icon" target="_blank">Graphite</a>) 👈 <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/7639?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `canary`
<hr />

## How to use the Graphite Merge Queue

Add either label to this PR to merge it via the merge queue:
- <kbd>merge</kbd> - adds this PR to the back of the merge queue
- <kbd>hotfix</kbd> - for urgent hot fixes, skip the queue and merge this PR next

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.dev/invite/github/toeverything?ref=merge-queue-instructions-in-stack-comment&prId=7639).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: packages/common/nbstore/src/storage/history.ts [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+    this.on('snapshot', snapshot => {
+      this.createHistory(snapshot.docId, snapshot).catch(noop);
+    });
+  }
+
+  override async setDocSnapshot(
+    snapshot: DocRecord,
+    prevSnapshot: DocRecord | null
+  ): Promise<boolean> {
+    const success = await this.setDocSnapshot(snapshot, prevSnapshot);
```
Comment (pengx17) [inline packages/common/nbstore/src/storage/history.ts]: super.xxx

Comment (forehalo) [reply packages/common/nbstore/src/storage/history.ts]: should be `this.upsertDocSnapshot `

Comment (forehalo) [reply packages/common/nbstore/src/storage/history.ts]: super.set is abstract function, delegate to another function is the only way to do this

**Review by EYHN — APPROVED** [review]

**Pull request merged**